packages/bundler-sdk-viem/src/operations.ts

```diff
@@ -326,18 +326,41 @@ export const populateSubBundle = (
     ) {
       // Liquidity is insufficient: trigger a public reallocation and try to have a resulting utilization as low as possible, above the target.
       // Solve: newTotalBorrowAssets / (newTotalSupplyAssets + reallocatedAssets) = supplyTargetUtilization
-      // Worst case is: there is not enough withdrawals available to fill reallocatedAssets, so utilization is above supplyTargetUtilization.
+      // We first try to find public reallocations that respect every markets targets.
+      // If this is not enough, the first market to be pushed above target is the supply market. Then we fully withdraw from every market.
       let requiredAssets =
         supplyTargetUtilization === 0n
           ? MathLib.MAX_UINT_160
           : MathLib.wDivDown(newTotalBorrowAssets, supplyTargetUtilization) -
             newTotalSupplyAssets;
 
-      const { withdrawals } = data.getMarketPublicReallocations(
+      let { withdrawals } = data.getMarketPublicReallocations(
         market.id,
         publicAllocatorOptions,
       );
 
+      const asssetsWithdrawn = withdrawals.reduce(
+        (total, { assets }) => total + assets,
+        0n,
+      );
+
+      if (
+        MarketUtils.getUtilization({
+          totalSupplyAssets: newTotalSupplyAssets + asssetsWithdrawn,
+          totalBorrowAssets: newTotalBorrowAssets,
+        }) > MathLib.WAD
```

Comment:
```suggestion
        newTotalSupplyAssets + asssetsWithdrawn < newTotalBorrowAssets
```

IMO it's much easier to reason this way here

Reply:
True

Reply:
But working with utilizations and `WAD` helps to understand the reasoning (about the 100% utilization)